src/extensions/step_handlers/swupdate_handler_v2/src/swupdate_handler_v2.cpp

```diff
@@ -185,9 +188,24 @@ ADUC_Result SWUpdateHandler_PerformAction(
     aduShellArgs.emplace_back(scriptFilePath);
     commandLineArgs.emplace_back(scriptFilePath);
 
-    aduShellArgs.emplace_back(adushconst::target_options_opt);
-    aduShellArgs.emplace_back(action.c_str());
-    commandLineArgs.emplace_back(action.c_str());
+    // Prepare arguments based on specified api version.
+    if (apiVer == nullptr || strcmp(apiVer, "2.0") == 0)
```

Comment:
Maybe it needs more offline discussion, but I don't think we should have code for older shell script api. There is contract version that is tied with handler version (e.g. microsoft/swupdate:2), so it seems reasonable to only support latest "shell script arg passing" contract/apiVer; otherwise, test matrix is multiplied.  Thinking is it should be ok as long as captured in release notes as a breaking change so that script authoring can be changed (will be tested before deploying new agent to production devices).

Reply:
Based on our previous offline discussions, this PR represents a significant enhancement for the libmicrosoft_script_handler v1. Customers now have the option to utilize the new features by including "apiVer":"2.1" in the import manifest. If the apiVer is not specified, customers will continue to use the script handler v1 as they did before. This update provides increased flexibility and improved functionality for our customers.